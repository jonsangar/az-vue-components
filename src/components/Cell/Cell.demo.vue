<template lang="pug">
  .cellDemo
    Demo(title="Uso BÃ¡sico")
      az-cell-group
        az-cell(title="Cell title" value="Content")
        az-cell(title="Cell title" value="Content" label="Description")

    Demo(title="Iconos")
      az-cell-group
        az-cell(title="Cell title" value="Content" left-icon="palm")
        az-cell(title="Cell title" value="Content" right-icon="palm")

    Demo(title="Solo contenido")
      az-cell-group
        az-cell(value="Content")

    Demo(title="Link")
      az-cell-group
        az-cell(title="Cell title" is-link)
        az-cell(title="Cell title" value="Content" is-link )
        az-cell(title="Cell title" value="Content" arrow-direction="down" is-link)

    Demo(title="Router")
      az-cell-group
        az-cell(title="Cell title" url="https://vuejs.org/" is-link)
        az-cell(title="Cell title" url="/packages" is-link )

    Demo(title="Uso de slots")
      az-cell-group
        az-cell(value="Content" is-link)
          template(v-slot:title)
            h4 Title slot
</template>

<script>
export default {};
</script>

<style lang="scss">
.cellDemo {
  min-height: 100vh;
  background-color: color(gray7);

  .demo__content {
    padding: 0;
  }
}
</style>
