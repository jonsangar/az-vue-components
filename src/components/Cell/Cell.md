---
device: true
---

# Celda

### Instalación

```javascript
import { Cell } from 'didor';

Vue.use(Cell);
```

---

### Uso

#### Uso básico

``` html
<az-cell-group>
  <az-cell title="Cell title" value="Content"/>
  <az-cell title="Cell title" value="Content" label="Description"/>
</az-cell-group>
```

#### Iconos

``` html
<az-cell-group>
  <az-cell title="Cell title" value="Content" left-icon="contact"/>
  <az-cell title="Cell title" value="Content" right-icon="contact"/>
</az-cell-group>
```

#### Solo contenido

``` html
<az-cell-group>
  <az-cell value="Content"/>
</az-cell-group>
```

#### Link

``` html
<az-cell-group>
  <az-cell title="Cell title" is-link/>
  <az-cell title="Cell title" value="Content" is-link/>
  <az-cell title="Cell title" value="Content" arrow-direction="down" is-link/>
</az-cell-group>
```

#### Router

``` html
<az-cell-group>
  <az-cell title="Cell title" url="https://vuejs.org/" is-link/>
  <az-cell title="Cell title" to="index" is-link/>
</az-cell-group>
```

#### Uso de slots

``` html
<az-cell-group>
  <az-cell value="Content" is-link>
    <template v-slot:title>
      <h4>Title slot</h4>
    </template>
  </az-cell>
</az-cell-group>
```

---

### API

| Attribute   | Description                                | Type      | Default   |
| ----------- | ------------------------------------------ | --------- | --------- |
| title | Título del campo | `String` `Number` | -- |
| value | Contenido del campo | `String` `Number` | -- |
| label | Descripción debajo del título | `String` | -- |
| title-width | Anchura de la etiqueta | `String` | `90px` |
| title-align | Alineación del texto de la etiqueta<br>`left` `center` `right` | `String` | `left` |
| value-align | Alineación del texto del input<br>`left` `center` `right` | `String` | `left` |
| left-icon | Icono del lado izquierdo | `String` | -- |
| right-icon | Icono del lado derecho | `String` | -- |
| url | Link url | `String` | -- |
| to | Target route, utiliza vue-roueter | `String` `Object` | -- |
| isLink | Muestra una flecha | `Boolean` | `false` |
| arrowDirection | Indica la dirección de la flecha<br>`left` `up` `right` `down` | `String` | `right` |

### Eventos

| Evento | Descripción                                             | Argumento |
| ------ | ------------------------------------------------------- | --------- |
| click | Se lanza cuando se hace click sobre la celda | `Event`         |

### Slots

| Nombre | Descripción                                             |
| ------ | ------------------------------------------------------- |
| title | Título del campo |
| label | Descripción del campo |
| default | Contenido del campo |
| leftIcon | Botón izquierdo |
| rightIcon | Botón derecho |
