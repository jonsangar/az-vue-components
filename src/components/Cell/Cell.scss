.az-cell {
  $rootClass: &;

  position: relative;
  font-size: font-size(base);
  line-height: simple();
  color: color(content);



  &__title {
    flex: none;
    // overflow: hidden;
    line-height: simple();
    // text-overflow: ellipsis;
    // white-space: nowrap;
  }

  &__value {
    position: relative;
    flex: 1;
    // overflow: hidden;
    color: color(gray4);
    vertical-align: middle;
  }

  &__label {
    font-size: font-size(small1);
    color: color(gray4);
  }

  &__body {
    display: flex;
    align-items: center;
  }

  &__left-icon,
  &__right-icon,
  &__link-icon {
    svg {
      width: simple();
      height: simple();
      padding: 0 0.1rem 0.1rem;
      margin: 0;
      color: color(gray4);
    }
  }

  &__left-icon {
    margin-right: 0.2rem;
    margin-left: quarter('-');

    svg {
      color: color(content);
    }
  }

  &__right-icon {
    margin-right: quarter('-');
    margin-left: 0.2rem;
  }

  &__link-icon {
    min-width: simple();
    margin-right: quarter('-');
    margin-left: 0.2rem;
    text-align: center;
    transition: all 0.5s cubic-bezier(0.075, 0.82, 0.165, 1);

    &--active {
      transform: rotate(90deg);
    }

    svg {
      color: color(brand);
    }
  }

  &--link {
    cursor: pointer;

    &:active {
      background-color: color(gray7);
    }
  }

  &--header{
    background-color: color(gray7);

    .az-cell__title {
      color: color(gray1);
    }

    .az-cell__value {
      color: color(gray3);
    }
  }

  &--subcell{
    background-color: color(gray8);
  }
  &--showError {
    padding-bottom: calc(#{simple()} + #{quarter()}) !important;
  }

  &__errorMessage {
    position: absolute;
    bottom: quarter();
    font-size: font-size(small1);
    color: color(danger);
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    width: calc(100% - #{simple()});

    &::before{
      content: '*';
    }
  }

}

@include media(palm) {
  .az-cell {
    display: flex;
    width: 100%;
    // padding: half() half();
    overflow: hidden;
    background-color: #ffffff;

    &--header{
      background-color: color(gray6);
    }

    &--subcell{
      background-color: color(gray8);
    }

    &:not(:last-child)::after {
      position: absolute;
      right: 0;
      bottom: 0;
      left: quarter();
      pointer-events: none;
      content: ' ';
      border-bottom: 1px solid color(border);
      -webkit-transform: scaleY(0.5);
      transform: scaleY(0.5);
    }
  }
}
