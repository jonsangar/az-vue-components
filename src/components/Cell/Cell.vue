<template lang="pug">
  .az-cell(:class="classObject" @click="onClick")
    //- Left Icon Block
    .az-cell__left-icon(
      v-if="leftIcon || $slots.leftIcon"
      @click="clickLeftIcon")
      slot(name="leftIcon")
        az-icon(
          v-if="scaleLeftIcon"
          :name="leftIcon"
          :scale="scaleLeftIcon"
          :color="leftIconColor")
        az-icon(:name="leftIcon" :color="leftIconColor" v-else)

    //- Title Block
    .az-cell__title(
      :style="{width: titleWidth, 'text-align': titleAlign}"
      v-if="title || $slots.title || label || $slots.label"
      @click="clickTitle")
      slot(name="title")
        | {{ title }}
        az-icon.margin-left-half(
            v-if="infoMessage && infoTextValidator(infoMessage)"
            name="icon-info"
            color="var(--color-gray6)"
            scale=".8"
          )
      .az-cell__label(v-if="label || $slots.label")
        slot(name="label")
          | {{ label }}

    //- Input block
    .az-cell__value
      .az-cell__body(:style="{'justify-content': valueAlignFlex}")
        slot
          | {{ value }}

        //- Link Icon Block
        .az-cell__link-icon(
          :class="{'az-cell__link-icon--active': linkActive}"
          v-if="isLink && !$slots.rightIcon")
          az-icon(:name="arrowDirection" :scale="arrowScale")

        //- Right Icon Block
        .az-cell__right-icon(
          v-if="(!isLink && rightIcon) || $slots.rightIcon"
          @click="clickRightIcon")
          slot(name="rightIcon")
            az-icon(:name="rightIcon" :color="rightIconColor")

    .az-cell__errorMessage(
      v-if="errorMessage")
      | {{ errorMessage }}

    PopupModal(
      v-if="showInfoPopup"
      :title="infoTitle"
      :message="infoMessage"
      :closeButton="true"
      text-align="left"
      @close="showInfoPopup = false")

</template>

<script>
import openLink from 'Core/utils/openLink.utils';
import PopupModal from 'Core/components/PopupModal';

export default {
  components: {
    PopupModal
  },

  props: {
    title: {
      type: String,
      required: false,
    },
    titleWidth: {
      type: [String, Number],
      default: '90px',
    },
    titleAlign: {
      type: String,
      default: 'left',
      validator: value => ['left', 'center', 'right'].indexOf(value) !== -1,
    },
    label: {
      type: String,
      required: false,
    },
    value: {
      type: [String, Number],
      required: false,
    },
    valueAlign: {
      type: String,
      default: 'right',
      validator: value => ['left', 'center', 'right'].indexOf(value) !== -1,
    },
    leftIcon: {
      type: String,
      required: false,
    },
    leftIconColor: {
      type: String,
      default: ''
    },
    scaleLeftIcon: {
      type: String,
      required: false,
    },

    rightIcon: {
      type: String,
      required: false,
    },
    rightIconColor: {
      type: String,
      default: ''
    },
    url: {
      type: String,
      required: false,
    },
    to: {
      type: [String, Object],
      required: false,
    },
    isLink: {
      type: Boolean,
      default: false,
    },
    linkActive: {
      type: Boolean,
      default: false,
    },
    type: {
      type: String,
      default: '',
      validator: value => ['header', 'subcell', ''].indexOf(value) !== -1
    },
    arrowDirection: {
      type: String,
      default: 'right',
      validator: value => ['left', 'right', 'up', 'down'].indexOf(value) !== -1,
    },

    errorMessage: {
      type: String,
      default: ''
    },

    infoMessage: {
      type: String,
      default: ''
    },

    infoTitle: {
      type: String,
      default: ''
    }
  },
  computed: {
    /**
     * Devuelve un listado con las clases que dependen de la configuración
     */
    classObject() {
      return [this.isLink ? 'az-cell--link' : '', this.type === 'header' ? 'az-cell--header' : '', this.type === 'subcell' ? 'az-cell--subcell' : '', this.errorMessage ? 'az-cell--showError' : ''];
    },

    valueAlignFlex() {
      if (!this.title && !this.$slots.title && !this.label && !this.$slots.label) {
        return 'flex-start';
      }

      switch (this.valueAlign) {
        case 'right':
          return 'flex-end';
        case 'center':
          return 'center';
        default:
          return 'flex-start';
      }
    },

    arrowScale() {
      return this.arrowDirection === 'up' || this.arrowDirection === 'down' ? '1' : '1.2';
    },

  },

  data() {
    return {
      showInfoPopup: false
    }
  },

  methods: {
    /**
     * Evento cuando se hace click en la celda
     */
    onClick(event) {
      if (this.url) {
        openLink(this.url);
        return;
      }

      if (this.to) {
        this.$router.push(this.to).catch(() => {});
        return;
      }

      this.$emit('click', event);
    },

    clickLeftIcon(event) {
      this.$emit('clickLeftIcon', event);
    },
    clickRightIcon(event) {
      this.$emit('clickRightIcon', event);
    },
    clickTitle(event) {
      if(this.infoMessage && this.infoTextValidator(this.infoMessage)) {
        this.showInfoPopup = true
      }
      this.$emit('clickTitle', event);
    },
    /**
     * Sólo valida textos que son incluídos en el archivo de traducción info.json siguiendo el formato: info.[key].desc
     * Si no existe la key en el archivo, no se va a mostrar el icono y el cuadro de info
     *
     * NOTA: Si el texto introducido en info proviene de otro archivo o formato no servirá esta validación
     */
    infoTextValidator(text) {
      if(text.startsWith('info.') && text.endsWith('.desc')) return false

      return true
    }
  },
};
</script>

<style src="./Cell.scss" lang="scss"></style>
