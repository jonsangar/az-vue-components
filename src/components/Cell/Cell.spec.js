import { mount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Cell from './Cell.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('Cell.vue', () => {
  const build = () => {
    const wrapper = mount(Cell, {
      localVue,
    })

    return {
      wrapper,
      cell: wrapper.find('.az-cell__body'),
    }
  }

  it('Comprueba renderizado del componente Cell con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que emite evento "click" al ser pulsado', async () => {
    const { wrapper } = build();
    // Al pulsar sobre el elemento lanza el evento click
    wrapper.trigger('click');
    // Esperamos para que emits sea manejado
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha emitido el evento click
    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
  });

  it('Comprueba que el botón tiene la propiedad isLink activada', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad isLink, añade la clase az-button--isLink
    await wrapper.setProps({
      isLink: true,
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.isLink).toStrictEqual(true);
    // Comprobamos que se ha añadido la clase isLink
    expect(wrapper.find('.az-cell--link').isVisible()).toBe(true);
  });

  it('Comprueba que el botón tiene la propiedad valueAlign a la izquierda', async () => {
    const { wrapper, cell } = build();
    // Si actualizamos la propiedad valueAlign, añade el valor flex-start al estilo justify-content
    await wrapper.setProps({
      valueAlign: 'left',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.valueAlign).toStrictEqual('left');
    expect(wrapper.html()).toMatchSnapshot()
    // Comprobamos que se ha añadido el valor flex-start al estilo justify-content
    expect(cell.element.style.getPropertyValue('justify-content')).toBe('flex-start');
  });

  it('Comprueba que el botón tiene la propiedad valueAlign a la derecha', async () => {
    const { wrapper , cell } = build();
    // Si actualizamos la propiedad valueAlign, añade el valor flex-end al estilo justify-content
    await wrapper.setProps({
      title: 'myTitle',
      valueAlign: 'right',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.valueAlign).toStrictEqual('right');
    // Comprobamos que se ha añadido el valor flex-end al estilo justify-content
    expect(cell.element.style.getPropertyValue('justify-content')).toBe('flex-end');
  });

  it('Comprueba que el botón tiene la propiedad valueAlign al centro', async () => {
    const { wrapper, cell } = build();
    // Si actualizamos la propiedad valueAlign, añade el valor center al estilo justify-content
    await wrapper.setProps({
      title: 'myTitle',
      valueAlign: 'center',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.valueAlign).toStrictEqual('center');
    // Comprobamos que se ha añadido el valor center al estilo justify-content
    expect(cell.element.style.getPropertyValue('justify-content')).toBe('center');
  });

  it('Comprueba que tiene el icono de la izquierda', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad leftIcon, añade el icono
    await wrapper.setProps({
      leftIcon: 'nest',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.leftIcon).toStrictEqual('nest');
    // Comprobamos que se ha añadido el icono
    expect(wrapper.find('.az-cell__left-icon').isVisible()).toBe(true);
  });

  it('Comprueba que tiene el icono de la izquierda con escalado', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad leftIcon, añade el icono con scale
    await wrapper.setProps({
      leftIcon: 'nest',
      scaleLeftIcon: '1',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
        expect(wrapper.html()).toMatchSnapshot()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.leftIcon).toStrictEqual('nest');
    expect(wrapper.vm.scaleLeftIcon).toStrictEqual('1');
    expect(wrapper.find('.az-cell__left-icon').isVisible()).toBe(true);
    // expect(wrapper.find('.az-icon').isVisible()).toBe(true);
    // Comprobamos que se ha añadido el icono con scale
    // expect(wrapper.find('.az-cell__link-icon').attributes().scale).toStrictEqual('1');
  });

  it('Comprueba que tiene el titulo', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad title, añade el título
    await wrapper.setProps({
      title: 'myTitle',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.title).toStrictEqual('myTitle');
    // Comprobamos que se ha añadido el titulo con estilos por defecto
    expect(wrapper.find('.az-cell__title').isVisible()).toBe(true);
    expect(wrapper.find('.az-cell__title').text()).toBe('myTitle');
    expect(wrapper.find('.az-cell__title').element.style.getPropertyValue('width')).toStrictEqual('90px');
    expect(wrapper.find('.az-cell__title').element.style.getPropertyValue('text-align')).toStrictEqual('left');
  });

  it('Comprueba que tiene el titulo con estilos', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad leftIcon, añade el icono con scale
    await wrapper.setProps({
      title: 'myTitle',
      titleWidth: '30%',
      titleAlign: 'center',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.title).toStrictEqual('myTitle');
    // Comprobamos que se ha añadido el titulo con estilos
    expect(wrapper.find('.az-cell__title').isVisible()).toBe(true);
    expect(wrapper.find('.az-cell__title').text()).toBe('myTitle');
    expect(wrapper.find('.az-cell__title').element.style.getPropertyValue('width')).toStrictEqual('30%');
    expect(wrapper.find('.az-cell__title').element.style.getPropertyValue('text-align')).toStrictEqual('center');
  });

  it('Comprueba que tiene el titulo con etiqueta', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad title y label, añade el título y label
    await wrapper.setProps({
      title: 'myTitle',
      label: 'myLabel'
    });
    // Esperamos para que se actualice las props
    await wrapper.vm.$nextTick()
    // Comprobamos que se han activado la props
    expect(wrapper.vm.title).toStrictEqual('myTitle');
    // Comprobamos que se ha añadido el titulo y label
    expect(wrapper.find('.az-cell__title').isVisible()).toBe(true);
    expect(wrapper.find('.az-cell__label').isVisible()).toBe(true);
    expect(wrapper.find('.az-cell__label').text()).toBe('myLabel');
  });

  it('Comprueba que tiene el icono de la derecha', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad rightIcon, añade el icono de la derecha
    await wrapper.setProps({
      rightIcon: 'rightArrow',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
    // Comprobamos que se han activado la prop
    expect(wrapper.vm.rightIcon).toStrictEqual('rightArrow');
    // Comprobamos que se ha añadido el icono con su name right
    expect(wrapper.find('.az-cell__right-icon').isVisible()).toBe(true);
    // expect(wrapper.find('.az-icon').isVisible()).toBe(true);
    // expect(wrapper.find('.az-cell__link-icon').attributes().name).toBe('rightArrow');
  });

  it('Comprueba que tiene el icono con dirección y escalado', async () => {
    const { wrapper } = build();
    // Si actualizamos las propiedades arrowDirection, añade el icono con escalado
    await wrapper.setProps({
      arrowDirection: 'up',
      isLink: true,
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
    // Comprobamos que se han activado la prop
    expect(wrapper.vm.arrowDirection).toStrictEqual('up');
    // Comprobamos que se ha añadido el icono con su name y escalado
    expect(wrapper.find('.az-cell__link-icon').isVisible()).toBe(true);
    // expect(wrapper.find('.az-icon').isVisible()).toBe(true);
    // expect(wrapper.find('.az-cell__link-icon').attributes().name).toBe('up');
    // expect(wrapper.find('.az-cell__link-icon').attributes().scale).toBe('1');
  });

  it('Comprueba el validator para la alineación del título "titleAlign"', async () => {
    // NOTA: Los validadores es mejor testearlos de forma aislada (sin montar);
    const validator = Cell.props.titleAlign.validator;

    // Testeamos el validador de la prop como una función normal
    expect(validator('left')).toBe(true);
    expect(validator('right')).toBe(true);
    expect(validator('center')).toBe(true);
    expect(validator('fakealign')).toBe(false);
  });

  it('Comprueba el validator para la alineación del valor "valueAlign"', async () => {
    // NOTA: Los validadores es mejor testearlos de forma aislada (sin montar);
    const validator = Cell.props.valueAlign.validator;

    // Testeamos el validador de la prop como una función normal
    expect(validator('left')).toBe(true);
    expect(validator('right')).toBe(true);
    expect(validator('center')).toBe(true);
    expect(validator('fakealign')).toBe(false);
  });

  it('Comprueba el validator para la dirección de flecha "arrowDirection"', async () => {
    // NOTA: Los validadores es mejor testearlos de forma aislada (sin montar);
    const validator = Cell.props.arrowDirection.validator;

    // Testeamos el validador de la prop como una función normal
    expect(validator('left')).toBe(true);
    expect(validator('right')).toBe(true);
    expect(validator('up')).toBe(true);
    expect(validator('down')).toBe(true);
    expect(validator('fakedirection')).toBe(false);
  });
})
