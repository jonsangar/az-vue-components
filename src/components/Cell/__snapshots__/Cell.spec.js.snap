// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell.vue Comprueba que el botón tiene la propiedad valueAlign a la izquierda 1`] = `
<div class="az-cell">
  <!---->
  <!---->
  <div class="az-cell__value">
    <div class="az-cell__body" style="justify-content: flex-start;">
      <!---->
      <!---->
    </div>
  </div>
  <!---->
</div>
`;

exports[`Cell.vue Comprueba que tiene el icono con dirección y escalado 1`] = `
<div class="az-cell az-cell--link">
  <!---->
  <!---->
  <div class="az-cell__value">
    <div class="az-cell__body" style="justify-content: flex-start;">
      <div class="az-cell__link-icon"><svg version="1.1" viewBox="0 0 16 16" class="az-icon az-fill"></svg></div>
      <!---->
    </div>
  </div>
  <!---->
</div>
`;

exports[`Cell.vue Comprueba que tiene el icono de la derecha 1`] = `
<div class="az-cell">
  <!---->
  <!---->
  <div class="az-cell__value">
    <div class="az-cell__body" style="justify-content: flex-start;">
      <!---->
      <div class="az-cell__right-icon"><svg version="1.1" viewBox="0 0 16 16" class="az-icon az-fill"></svg></div>
    </div>
  </div>
  <!---->
</div>
`;

exports[`Cell.vue Comprueba que tiene el icono de la izquierda con escalado 1`] = `
<div class="az-cell">
  <div class="az-cell__left-icon"><svg version="1.1" viewBox="0 0 16 16" class="az-icon az-fill"></svg></div>
  <!---->
  <div class="az-cell__value">
    <div class="az-cell__body" style="justify-content: flex-start;">
      <!---->
      <!---->
    </div>
  </div>
  <!---->
</div>
`;

exports[`Cell.vue Comprueba renderizado del componente Cell con snapshot 1`] = `
<div class="az-cell">
  <!---->
  <!---->
  <div class="az-cell__value">
    <div class="az-cell__body" style="justify-content: flex-start;">
      <!---->
      <!---->
    </div>
  </div>
  <!---->
</div>
`;
